src/billing/Project.ts

```diff
@@ -1,33 +1,49 @@
 import { SpinnerOverlay } from '@terra-ui-packages/components';
 import _ from 'lodash/fp';
 import * as qs from 'qs';
-import { Fragment, useEffect, useMemo, useState } from 'react';
+import { Fragment, ReactNode, useEffect, useMemo, useState } from 'react';
 import { div, h, p, span } from 'react-hyperscript-helpers';
 import { BillingAccountControls } from 'src/billing/BillingAccount/BillingAccountControls';
-import { BillingAccountSummary } from 'src/billing/BillingAccount/BillingAccountSummary';
+import { BillingAccountSummary, BillingAccountSummaryProps } from 'src/billing/BillingAccount/BillingAccountSummary';
 import { Members } from 'src/billing/Members/Members';
 import { ExternalLink } from 'src/billing/NewBillingProjectWizard/StepWizard/ExternalLink';
 import { SpendReport } from 'src/billing/SpendReport/SpendReport';
-import { accountLinkStyle, billingRoles } from 'src/billing/utils';
+import { accountLinkStyle, BillingAccountStatus, billingRoles } from 'src/billing/utils';
 import { Workspaces } from 'src/billing/Workspaces/Workspaces';
 import { Link } from 'src/components/common';
+import { User } from 'src/components/group-common';
 import { icon } from 'src/components/icons';
 import { InfoBox } from 'src/components/InfoBox';
 import { SimpleTabBar } from 'src/components/tabBars';
 import { Ajax } from 'src/libs/ajax';
-import { isAzureBillingProject, isGoogleBillingProject } from 'src/libs/ajax/Billing';
+import {
+  BillingProject,
+  GoogleBillingAccount,
+  isAzureBillingProject,
+  isGoogleBillingProject,
+} from 'src/libs/ajax/Billing';
 import colors from 'src/libs/colors';
 import { reportErrorAndRethrow } from 'src/libs/error';
 import Events, { extractBillingDetails } from 'src/libs/events';
 import * as Nav from 'src/libs/nav';
 import { useCancellation, useGetter, useOnMount, usePollingEffect } from 'src/libs/react-utils';
 import * as StateHistory from 'src/libs/state-history';
 import * as Utils from 'src/libs/utils';
+import { isGoogleWorkspaceInfo, WorkspaceInfo, WorkspaceWrapper } from 'src/workspaces/utils';
 
-const groupByBillingAccountStatus = (billingProject, workspaces) => {
-  const group = (workspace) =>
+export const groupByBillingAccountStatus = (
+  billingProject: BillingProject,
+  workspaces: WorkspaceInfo[]
+): Record<BillingAccountStatus, Set<WorkspaceInfo>> => {
+  const group = (workspace: WorkspaceInfo): BillingAccountStatus =>
     Utils.cond(
-      [billingProject.billingAccount === workspace.billingAccount, () => 'done'],
+      [isAzureBillingProject(billingProject), () => 'done'],
```

Comment:
This was always true because for Azure billing projects, `billingProject.billingAccount === undefined`, as does `workspace.billingAccount`. But those variables don't exist on Azure billing projects and workspaces, so for TypeScript we need to be explicit about the condition.